src/frontend/src/app/AppPublic.tsx

```diff
@@ -17,7 +17,11 @@ const AppPublic: React.FC = () => {
 
   const devUserId = localStorage.getItem('devUserId');
 
-  const render = (e: any) => {
+  interface AppRenderInput {
```

Comment:
ah sorry didn't realize this before but this should go before the component declaration